### PR TITLE
PR: Add save question for modified untitled notebooks before closing them

### LINKED ISSUE #64: Ask users if they want to save untitled notebooks before closing them
This is an important issue I didn't realize before. Just like for the Editor, we need to ask users if they want to save untitled notebooks before closing them, unless (obviously) they are empty.

@dalthviz, please take care of this one.

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #64

### CODE REVIEW

**Comment by dalthviz** [comment]
Preview:

saveuntitled

**Comment by goanpeca** [comment]
@dalthviz looks great, any chance we can disable (hide) the autosaving dialog that appears, it is kind of misleading within the context of spyder

**Comment by dalthviz** [comment]
@goanpeca I will try to find a way to hide the autosaving dialog :+1:

**Comment by dalthviz** [comment]
Preview without the dialog:

saveuntitled

File: spyder_notebook/notebookplugin.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def create_new_client(self, filename=None, give_focus=True):
         client.register(server_info)
         client.load_notebook()
 
-    def close_client(self, index=None, client=None):
+    def close_client(self, index=None, client=None, saving=False):
```
Comment (ccordoba12) [inline spyder_notebook/notebookplugin.py]: Please change `saving` to `save` here.

File: spyder_notebook/notebookplugin.py [hunk 2]
```diff
@@ -242,15 +242,34 @@ def close_client(self, index=None, client=None):
...
+
         client.shutdown_kernel()
         client.close()
 
         # Note: notebook index may have changed after closing related widgets
         self.tabwidget.removeTab(self.tabwidget.indexOf(client))
         self.clients.remove(client)
 
-    def save_as(self, name=None):
+    def save_as(self, name=None, closing=False):
```
Comment (ccordoba12) [inline spyder_notebook/notebookplugin.py]: Please change `closing` to `close`

**Comment by ccordoba12** [comment]
I made a couple of simple suggestions, but this looks so cool!! Thanks @dalthviz for coming up with a solution for my suggestion.

**Comment by dalthviz** [comment]
@ccordoba12  I think this is ready to merge.

**Comment by ccordoba12** [comment]
Thanks @dalthviz!

**Pull request merged**